### ISSUE TITLE
[feature request]? Overlay evaluation results on jupyter-scratch buffer

### ISSUE THREAD

**Issue by dangom** [dzop/emacs-jupyter#143]
In similar spirit to [cider overlays](https://github.com/clojure-emacs/cider/blob/master/cider-overlays.el) and [Emacs Result OverlayS](https://github.com/xiongtx/eros), would it be make sense to have python evaluation results overlayed briefly when evaluating a line with `C-x C-e` in the jupyter-scratch buffer?

**Comment by dzop** [comment]
I have been working on this feature. Its on the [feature/eval-overlays](https://github.com/dzop/emacs-jupyter/tree/feature/eval-overlays) branch. If you want to try it out you will have to set `jupyter-eval-use-overlays-p` to `t` to enable it after pulling in the changes. Any feedback would be appreciated.

**Comment by dangom** [comment]
Wow, amazing! Will do!

**Comment by dangom** [comment]
Overlays do work OK! No major issues.

The only small annoyances I found concern navigation and character deletion. The overlays seem to become part of the last character on a line, so motion to the end-of-line feels a bit awkward because it goes to the end of the overlay. Similar issues with `backward-char`.

Aesthetically, I would also prefer overlays a bit smaller and without the bounding box, but the branch already offers a `jupyter-eval-overlay` face, so that's a non-issue.

**Comment by dzop** [comment]
>  so motion to the end-of-line feels a bit awkward because it goes to the end of the overlay

Luckily I was able to quickly find the fix for this. We can just use the [cursor](https://www.gnu.org/software/emacs/manual/html_node/elisp/Special-Properties.html#Special-Properties) text property on the first character of the `display` string of the overlay. From its documentation:

> the string character with the `cursor` property of any non-`nil` value is the character where to display the cursor

I just force pushed to the branch some changes that use the `cursor` property and it seems to prevent moving past the overlay with the motion commands I've tried.

>  I would also prefer overlays a bit smaller and without the bounding box

Yeah I don't know what a good default face would be. I liked the bounding box because its a clearly delineates the overlay result from the rest of the text, but we could also do this with a different background color. The bounding box also looks ugly with multiline results since each line is boxed so I'll probably change it.
